eqcorrscan/core/match_filter/tribe.py

```diff
@@ -606,7 +606,21 @@ def detect(self, stream, threshold, threshold_type, trig_int, plot=False,
         if len(party) > 0:
             for family in party:
                 if family is not None:
-                    family.detections = family._uniq().detections
+                    # Slow uniq:
+                    # family.detections = family._uniq().detections
+                    # Very quick uniq:
+                    det_tuples = [
+                        (det.id, str(det.detect_time), det.detect_val)
+                        for det in family]
+                    # Retrieve the indices for the first occurrence of each
+                    # detection in the family (so only unique detections will
+                    # remain).
+                    uniq_det_tuples, uniq_det_indices = np.unique(
+                        det_tuples, return_index=True, axis=0)
+                    uniq_detections = []
+                    for uniq_det_index in uniq_det_indices:
+                        uniq_detections.append(family[uniq_det_index])
+                    family.detections = uniq_detections
```

Comment:
This should be fine (for what we want, which is no multiple detections at the same time for the same template and hence family, this is ideal) - and a lot faster! Nice!